src/kibali/OrderedPairEqualityComparer.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System.Collections.Generic;
+
+namespace Kibali;
+
+public class OrderedPairEqualityComparer : IEqualityComparer<(string, string)>
+{
+    public bool Equals((string, string) x, (string, string) y)
+    {
+        // Check if the items are equal regardless of order
+        return (x.Item1 == y.Item1 && x.Item2 == y.Item2) || (x.Item1 == y.Item2 && x.Item2 == y.Item1);
```

Comment:
Since we are comparing strings it may be better to be explicit on casing comparison e.g. `(x.Item1.Equals(y.Item1,StringComparison.OrdinalIgnoreCase)`. We could possibly have the comparison type as a constructor argument.

Reply:
Good catch. Updated